pkg/virtctl/ssh/ssh.go

```diff
@@ -0,0 +1,157 @@
+		fmt.Sprintf("--%s=/home/jdoe/.ssh/id_rsa: Set the path to a private key used for authenticating to the server; If not provided, the client will try to use the local ssh-agent at $SSH_AUTH_SOCK", identityFilePathFlag))
+	cmd.Flags().IntVarP(&sshPort, portFlag, portFlagShort, 22,
+		fmt.Sprintf(`--%s=22: Specify a port on the VM to send SSH traffic to`, portFlag))
+	cmd.Flags().BoolVar(&wrapLocalSSH, wrapLocalSSHFlag, wrapLocalSSH,
+		fmt.Sprintf("--%s=true: Set this to true to use the SSH command available on your system by using this command as ProxyCommand; If unassigned, this will establish a SSH connection with limited capabilities provided by this client", wrapLocalSSHFlag))
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	return cmd
+}
+
+type SSH struct {
+	clientConfig clientcmd.ClientConfig
+}
+
+func (o *SSH) Run(cmd *cobra.Command, args []string) error {
+	kind, namespace, name, err := o.prepareCommand(args)
+	if err != nil {
+		return err
+	}
+
+	if wrapLocalSSH {
+		return runLocalCommandClient(kind, namespace, name)
+	}
+
+	client, err := o.prepareSSHClient(kind, namespace, name)
+	if err != nil {
+		return err
+	}
+	return o.startSession(client)
+}
+
+func (o *SSH) prepareCommand(args []string) (kind, namespace, name string, err error) {
+	var targetUsername string
+	kind, namespace, name, targetUsername, err = templates.ParseSSHTarget(args[0])
+	if err != nil {
+		return
+	}
+
+	if len(namespace) < 1 {
+		namespace, _, err = o.clientConfig.Namespace()
+		if err != nil {
+			return
+		}
+	}
+
+	if len(sshUsername) > 0 && len(targetUsername) > 0 {
+		err = fmt.Errorf("only one userame must be specified, got --%s=%s and %s@",
+			usernameFlag, sshUsername, targetUsername)
+		return
+	}
+
+	if len(targetUsername) > 0 {
+		sshUsername = targetUsername
+	}
+
+	return
+}
+
+func (o *SSH) prepareSSHTunnel(kind, namespace, name string) (kubecli.StreamInterface, error) {
+	virtCli, err := kubecli.GetKubevirtClientFromClientConfig(o.clientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	var stream kubecli.StreamInterface
+	if kind == "vmi" {
+		stream, err = virtCli.VirtualMachineInstance(namespace).PortForward(name, sshPort, "tcp")
+		if err != nil {
+			return nil, fmt.Errorf("can't access VMI %s: %w", name, err)
+		}
+	} else if kind == "vm" {
+		stream, err = virtCli.VirtualMachine(namespace).PortForward(name, sshPort, "tcp")
+		if err != nil {
+			return nil, fmt.Errorf("can't access VM %s: %w", name, err)
+		}
+	}
+
+	return stream, nil
+}
+
+func usage() string {
+	return fmt.Sprintf(`  # Connect to 'testvmi' (using the built-in SSH client):
+  {{ProgramName}} ssh jdoe@testvmi [--%s]
+
+  # Connect to 'testvm' in 'mynamespace' namespace
+  {{ProgramName}} ssh jdoe@vm/testvm.mynamespace [--%s]
+
+  # Specify a username and namespace:
+  {{ProgramName}} ssh --namespace=mynamespace --%s=jdoe testvmi
+ 
+  # Connecto to 'testvmi' using your local OpenSSH:
```

Comment:
Connecto/Connect to 

also, `using your local OpenSSH` might make more sense as something like `using the local ssh binary found in $PATH`